x-pack/plugins/siem/public/components/navigation/index.test.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { shallow } from 'enzyme';
+import { LocationDescriptorObject, LocationListener, UnregisterCallback } from 'history';
+import * as React from 'react';
+
+import { SiemNavigationComponent } from './';
+import { setBreadcrumbs } from './breadcrumbs';
+
+jest.mock('./breadcrumbs', () => ({
+  setBreadcrumbs: jest.fn(),
+}));
+type Action = 'PUSH' | 'POP' | 'REPLACE';
+const pop: Action = 'POP';
+describe('SIEM Navigation', () => {
+  const location = {
+    pathname: '/hosts',
+    search: '',
+    state: '',
+    hash: '',
+  };
+  const mockProps = {
+    location,
+    match: {
+      isExact: true,
+      params: {},
+      path: '',
+      url: '',
+    },
+    history: {
+      length: 2,
+      location,
+      action: pop,
+      push: () => null,
+      replace: () => null,
```

Comment:
God this was obnoxious, was going back and forth with typescript to get this monster together. If anyone has a better way, please let me know!

Reply:
maybe something like that
```/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { shallow } from 'enzyme';
import { LocationDescriptorObject, LocationListener, UnregisterCallback } from 'history';
import * as React from 'react';

import { SiemNavigationComponent } from './';

jest.unmock('./breadcrumbs');
import * as myBreadcrumbs from './breadcrumbs';

type Action = 'PUSH' | 'POP' | 'REPLACE';
const pop: Action = 'POP';
describe('SIEM Navigation', () => {
  let mockSetBreadcrumbs = jest.fn();

  const location = {
    pathname: '/hosts',
    search: '',
    state: '',
    hash: '',
  };
  const mockProps = {
    location,
    match: {
      isExact: true,
      params: {},
      path: '',
      url: '',
    },
    history: {
      length: 2,
      location,
      action: pop,
      push: jest.fn(),
      replace: jest.fn(),
      go: jest.fn(),
      goBack: jest.fn(),
      goForward: jest.fn(),
      block: jest.fn().mockImplementation((t: UnregisterCallback) => t),
      createHref: jest.fn().mockImplementation((t: LocationDescriptorObject) => 't'),
      listen: jest.fn().mockImplementation((l: LocationListener) => {
        const temp: UnregisterCallback = () => null;
        return temp;
      }),
    },
  };

  beforeEach(() => {
    jest.clearAllMocks();
    mockSetBreadcrumbs = jest.fn();
    Object.defineProperty(myBreadcrumbs, 'setBreadcrumbs', { get: () => mockSetBreadcrumbs });
  });

  test('it calls setBreadcrumbs with correct path on mount', () => {
    shallow(<SiemNavigationComponent {...mockProps} />);

    expect(mockSetBreadcrumbs.mock.calls[0][0]).toEqual('/hosts');
  });
  test('it calls setBreadcrumbs with correct path on update', () => {
    const wrapper = shallow(<SiemNavigationComponent {...mockProps} />);
    wrapper.setProps({ location: { pathname: '/network' } });
    wrapper.update();

    expect(mockSetBreadcrumbs.mock.calls[1][0]).toEqual('/network');
  });
});
```

Reply:
If I'm not using all those mocks, doesn't it make more sense to just put that noop function that I had rather than creating unused functionality?

Reply:
It seems that we just have to make typescript happy and these mock value are not in used in the unit test. This pass as well.
```
history: {
      length: 0,
      location,
      action: pop,
      push: jest.fn(),
      replace: jest.fn(),
      go: jest.fn(),
      goBack: jest.fn(),
      goForward: jest.fn(),
      block: jest.fn(),
      createHref: jest.fn(),
      listen: jest.fn(),
    }, 
```

Reply:
alright i like that enough, thanks!